packages/webapp/src/components/Form/CompositionInputs/NumberInputWithSelect.tsx

```diff
@@ -0,0 +1,145 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import { ReactNode, useEffect } from 'react';
+import clsx from 'clsx';
+import { useTranslation } from 'react-i18next';
+import ReactSelect from '../ReactSelect';
+import useReactSelectStyles from '../Unit/useReactSelectStyles';
+import useNumberInput from '../NumberInput/useNumberInput';
+import InputBase from '../InputBase';
+import { styles as reactSelectDefaultStyles } from '../ReactSelect';
+import styles from './styles.module.scss';
+
+export type Option = { value: string; label: ReactNode };
+
+export const UNIT = 'unit';
+
+export enum NPK {
```

Comment:
Is NPK in this component temporarily until a container is made? 

Otherwise maybe it should be generic reusable or have a Component name like NPK input?

Reply:
Thank you for reviewing!
I think I was trying to make it reusable, but I couldn't handle types, and this is truly non-reusable component now...
I will revisit this again later when we work on some of those additional inputs, and if I can't reuse this, I will rename to NPKInput!